rocAL_pybind/amd/rocal/plugin/generic.py

```diff
@@ -33,6 +33,107 @@
+                if self.return_max_roi:
+                    self.num_dims = len(dimensions) - 1
+                    self.roi_array = np.zeros(self.batch_size * self.num_dims * 2, dtype=np.uint32)
+                    self.output_tensor_list[i].copy_roi(self.roi_array)
+                    self.max_roi_size = np.zeros(self.num_dims, dtype=np.uint32)
+                    for j in range(self.batch_size):
+                        index = j * self.num_dims * 2
+                        roi_size = self.roi_array[index + self.num_dims: index +
+                                                  self.num_dims * 2] - self.roi_array[index: index + self.num_dims]
+                        self.max_roi_size = np.maximum(roi_size, self.max_roi_size)
+                if self.device == "cpu":
+                    self.dtype = self.output_tensor_list[i].dtype()
+                    output = np.empty(
+                        dimensions, dtype=self.dtype)
+                else:
+                    with cp.cuda.Device(device=self.device_id):
+                        self.dtype = self.output_tensor_list[i].dtype()
+                        output = cp.empty(dimensions, self.dtype)
+
+                if self.device == "cpu":
+                    self.output_tensor_list[i].copy_data(output)
+                else:
+                    self.output_tensor_list[i].copy_data(output.data.ptr)
+                self.output_list.append(output)
+        else:
+            for i in range(len(self.output_tensor_list)):
+                if self.return_max_roi:
+                    self.output_tensor_list[i].copy_roi(self.roi_array)
+                    self.max_roi_size = np.zeros(self.num_dims, dtype=np.uint32)
+                    for j in range(self.batch_size):
+                        index = j * self.num_dims * 2
+                        roi_size = self.roi_array[index + self.num_dims: index +
+                                                  self.num_dims * 2] - self.roi_array[index: index + self.num_dims]
+                        self.max_roi_size = np.maximum(roi_size, self.max_roi_size)
+                if self.device == "cpu":
+                    self.output_tensor_list[i].copy_data(self.output_list[i])
+                else:
+                    self.output_tensor_list[i].copy_data(
+                        self.output_list[i].data.ptr)
+        if self.return_max_roi:
+            roi_output_list = []
+            for i in range(len(self.output_list)):
+                roi_output_list.append(
+                    self.output_list[i][:, :self.max_roi_size[0], :self.max_roi_size[1], :self.max_roi_size[2], :self.max_roi_size[3]])
+            # Check if last batch policy is partial and only return the valid images in last batch
+            if (self.last_batch_policy is (types.LAST_BATCH_PARTIAL)) and b.getRemainingImages(self.loader._handle) <= 0:
+                if (self.last_batch_size is None):
+                    self.last_batch_size = self.batch_size - \
```

Comment:
What is the \ for and can it not be done in same line?

Reply:
Autopep8 formatting